pulp_rpm/app/models.py

```diff
@@ -775,29 +846,89 @@ class Category(Content):
             A dictionary of names by language
         digest (Text):
             A checksum for the category
+        packagegroups (ManyToMany):
+            PackageGroups related to this category
     """
 
-    TYPE = 'category'
+    TYPE = 'packagecategory'
 
     # Required metadata
     id = models.CharField(max_length=255)
 
     name = models.CharField(max_length=255)
-    description = models.TextField()
-    display_order = models.IntegerField()
+    description = models.TextField(default='')
+    display_order = models.IntegerField(null=True)
 
     group_ids = JSONField(default=list)
 
-    desc_by_lang = models.TextField()
-    name_by_lang = models.TextField()
+    desc_by_lang = JSONField(default=dict)
+    name_by_lang = JSONField(default=dict)
 
     digest = models.CharField(unique=True, max_length=64)
 
+    packagegroups = models.ManyToManyField(PackageGroup)
+
     class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
 
+    @classmethod
+    def natural_key_fields(cls):
+        """
+        Digest is used as a natural key for PackageCategory.
+        """
+        return ('digest',)
+
+    @classmethod
+    def grplist_to_lst(cls, value):
+        """
+        Convert libcomps GrpList to list.
+
+        Args:
+            value: a libcomps GrpList
+
+        Returns:
+            A list
+
+        """
+        grp_list = []
+        if len(value):
```

Comment:
same question here, does it complain without this check?
For normal python types, there is no need to do so but I can imagine that libcomps type might be a special case :)